File: lista3/zadanie1/Szesciokat.java
public class Szesciokat extends Figura {

    private int bok;

    Szesciokat(int bok){
        if (bok < 0) throw new IllegalArgumentException(bok + "-> niedodatni bok :c");
        this.bok = bok;
    }

    public int dajBok(){
        return bok;
    }

    public double obliczPole(){
        return ( ( 3.0 / 2.0 ) * bok * bok * Math.sqrt(3.0)); 
    }

    public double obliczObwod(){
        return (6 * bok);
    }

    public void dajNazwe(){
        System.out.println("Jestem szesciokatem");
    }
}

File: lista3/zadanie2/Szesciokat.cpp
#include "Figura.hpp"
#include "Szesciokat.hpp"
#include <cmath>



double Szesciokat::obliczPole(){
    if (bok <= 0) 
        throw std::invalid_argument(std::to_string(bok) + std::string("-> Otrzymano niedodatni promien"));
    return (3/2.0) * bok * bok * sqrt(3.0);
}

double Szesciokat::obliczObwod(){
    if (bok <= 0) 
        throw std::invalid_argument(std::to_string(bok) + std::string("-> Otrzymano niedodatni bok"));
    return 6*bok;
}

void Szesciokat::dajNazwe(){
    std::cout << "Jestem szesciokatem" << std::endl;
}